### ISSUE TITLE
"Tried to detach xxx not in a location" prompts at random times in build 2023-11-04-0144

### ISSUE THREAD

**Issue by Unrepentant-Atheist** [cataclysmbnteam/Cataclysm-BN#3543]
### Describe the bug

Wanted to equip/wield a pipe directly from a tile nearby. These errors prompted. First time this prompted was when I tried to equip a pipe, no idea what boot_combat have to do with that.

Then this happend: https://github.com/cataclysmbnteam/Cataclysm-BN/issues/3543#issuecomment-1793389856

### Steps To Reproduce

1. Equip/wield a pipe
2. ???
3. Don't profit.

### Screenshots

![image](https://github.com/cataclysmbnteam/Cataclysm-BN/assets/3938423/f9b4ea8b-9910-49ff-b2ff-e24b0c58e6cf)

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19045.3570 (22H2)
- Game Version: 7ed3776 [64-bit]
- Graphics Version: Tiles
- LAPI Version: 1
- Game Language: English [en_US]
- Mods loaded: [
    Bright Nights [bn],
    SpeedyDex [speedydex],
    Stats Through Kills [stats_through_kills],
    Stats Through Skills [StatsThroughSkills],
    Bionic Slots [cbm_slots],
    Elevated bridges [elevated_bridges],
    Cataclysm++ [Cata++],
    Aftershock [aftershock],
    PKs Rebalancing [PK_REBALANCE],
    Secronom [secronom],
    Secronom+ Lore Expansion [secronom_lore_expansion],
    C.R.I.T. Expansion [crt_expansion],
    Sierra Madre [sierra_madre],
    Bio-Synth Robot expansion [gekkobots],
    Nechronica Redux [Nechronica_Redux],
    Magiclysm [magiclysm],
    Really Dark Skies [realdarkskies],
    Arcana and Magic Items [Arcana],
    Project: Pack Rat [Jake's_Squirrel_mod],
    Add Bandits Extended+ [GOV_BANDITS_KAI_R],
    NPC Increased Location Spawns Combined [npc_increased_location_comb],
    More Bandits, Testing version [More_Bandits],
    More Dangerous Encounters [More_Dangerous_Encounters],
    Vampiric Stuff [vamp_stuff],
    Vampiric Stuff + Arcana Mod Extension [vamp_stuff+arcana],
    Magiclysm Revamp and Additions [magicrevamp],
    Miscellaneous Magiclysm Expansions [demon_spider_a],
    Urban Development [Urban_Development],
    Morlock Village [morlock_village],
    Arcana/Cata++ Patchmod [Arcana_Cata++_Patch],
    Arcana/Aftershock Patchmod [Arcana_aftershock_Patch],
    Arcana/C.R.I.T. Patchmod [Arcana_Crit_Patch],
    Arcana/Magiclysm Patchmod [Arcana_Magiclysm_Patch],
    Really Dark Skies plus Arcana [realdarkskies_Arcana_patchmod],
    Really Dark Skies plus Cataclysm++ [realdarkskies_CATA_patchmod],
    Really Dark Skies plus Aftershock [realdarkskies_aftershock_patchmod],
    Really Dark Skies plus PKs Rebalancing [realdarkskies_PK_REBALANCE_patchmod],
    Really Dark Skies plus Secronom [realdarkskies_secronom_patchmod],
    Really Dark Skies plus C.R.I.T. [realdarkskies_CRIT_patchmod],
    Really Dark Skies plus Bio-Synth Robot Expansion [realdarkskies_gekkobots_patchmod],
    Really Dark Skies plus magiclysm [realdarkskies_magiclysm_patchmod],
    Really Dark Skies plus Morlock Village [realdarkskies_morlock_village_patchmod],

]

### Additional context

Debug.log
```
05:23:44.631 ERROR DEBUGMSG : D:\a\Cataclysm-BN\Cataclysm-BN\src\game_object.cpp:47 [detach] Tried to detach [boots_combat] not in a location.
05:27:31.241 ERROR DEBUGMSG : D:\a\Cataclysm-BN\Cataclysm-BN\src\detached_ptr.cpp:57 [get] Attempted to resolve invalid detached_ptr
05:27:33.124 ERROR DEBUGMSG : D:\a\Cataclysm-BN\Cataclysm-BN\src\detached_ptr.cpp:57 [get] Attempted to resolve invalid detached_ptr
[ Previous repeated 6 times ]
05:27:33.124 ERROR DEBUGMSG : D:\a\Cataclysm-BN\Cataclysm-BN\src\detached_ptr.cpp:57 [get] Attempted to resolve invalid detached_ptr
05:27:33.124 ERROR DEBUGMSG : D:\a\Cataclysm-BN\Cataclysm-BN\src\game_object.cpp:47 [detach] Tried to detach [box_cigarette] not in a location.
05:27:33.124 ERROR DEBUGMSG : D:\a\Cataclysm-BN\Cataclysm-BN\src\detached_ptr.cpp:57 [get] Attempted to resolve invalid detached_ptr
[ Previous repeated 7 times ]
05:27:33.124 ERROR DEBUGMSG : D:\a\Cataclysm-BN\Cataclysm-BN\src\detached_ptr.cpp:57 [get] Attempted to resolve invalid detached_ptr
```

**Comment by AniDemi** [comment]
1. Got a fresh install from releases
2. Spawned pipe, tried equipping it from the floor and from my inventory, no errors
3. Did the same with a pipe spawned from breaking a locker, no errors
I believe the issue to be either caused by the massive amount of mods you have or a improper install

**Comment by Unrepentant-Atheist** [comment]
Well, title needs to be changed. I just happend while I did nothing but was moving around (tried fleeing a burning build and a horde on me arse).

**Comment by AniDemi** [comment]
I randomly encountered this bug while I was testing and spawned an npc, I believe the error happened when it was trying to pick up items, but I find myself unable to replicate the bug when spawning another npc in another world.

**Comment by Unrepentant-Atheist** [comment]
Can be probably marked as verified: [trying to eat coffeinated chewing gum in newest version](https://discord.com/channels/830879262763909202/830916451517857894/1171147277969133729)

**Comment by joveeater** [comment]
It's hard to be sure but I think this was caused by an npc picking something up and is fixed by #3629.
